### ISSUE TITLE
Loading custom library fails

### ISSUE THREAD

**Issue by TLWebdesign** [vdm-io/Joomla-Component-Builder#399]
### Steps to reproduce the issue
Add library local file to JCB tell it to be placed in /media/ and add to site view. Compile and view site view. 

### Expected result
Files being added succesfully

### Actual result

Warning: require_once(): open_basedir restriction in effect. File(/media/com_***/OdrApi/Odr.php) is not within the allowed path(s): (/home/***/:/tmp/:/var/tmp/:/usr/local/php73/lib/:/usr/local/php54/lib/:/usr/local/php55/lib/:/usr/local/php56/lib/:/usr/local/php70/lib/:/usr/local/php71/lib/:/usr/local/php72/lib/:/usr/local/php73/lib/:/usr/local/lib/php/) in /home/***/domains/***.nl/public_html/dev/components/com_***/views/***/view.html.php on line 70

### System information (as much as possible)
- OS Name & Version: MacOS Mojave
- MySql Version: MariaDB 10.3
- Apache Version: 2.4
- PHP Version: 7.3
- Joomla Version: 3.9.4
- JCB Version: 2.9.11
- Browser:  Firefox 65.0.1

### Additional comments
Screenshots showing library settings and view.html.php.

![Schermafbeelding 2019-03-15 om 00 02 18](https://user-images.githubusercontent.com/4402824/54397274-3d6c2300-46b6-11e9-984e-adefdf5231bf.png)
![Schermafbeelding 2019-03-15 om 00 06 23](https://user-images.githubusercontent.com/4402824/54397275-3e04b980-46b6-11e9-9a6b-a2664250d647.png)

Adding JPATH_SITE like this:

		require_once(JPATH_SITE ."/media/com_***/OdrApi/Odr.php");
		require_once(JPATH_SITE ."/media/com_***/OdrApi/Odr/Exception.php"); 

seems to fix the issue.

**Comment by Llewellynvdm** [comment]
Okay libraries was originally added for JavaScript and CSS (libs) :+1: 

To also use it to add php stuff is not ideal, I would much rather custom code this into the view and add the files via the file and folder area.

But having said that, I suppose you could use libraries for anything... we may just have to rethink some of its defaults.

For now I think using the "Custom Script" option as "File Behaviour" will seem to resolve your issue.

To then implement your script you will add it like this:
```
	require_once(JPATH_SITE . "/components/com_[[[component]]]/helpers/OdrApi/Odr.php");
	require_once(JPATH_SITE . "/components/com_[[[component]]]/helpers/OdrApi/Odr/Exception.php");
```

I then tried to take some screenshots of how this should look, and then found there are some JavaScript issues in JCB that does not show the linked files and folders. So I will fix that!

But the above code and the using the Advance folder inclusion should work.

**Comment by Llewellynvdm** [comment]
Okay I fixed the JavaScript and will push up the fix soon... but here is the screen shots.
![image](https://user-images.githubusercontent.com/5607939/54556967-c63ed380-49c2-11e9-9dbe-34062ddccb02.png)

![image](https://user-images.githubusercontent.com/5607939/54557036-f0909100-49c2-11e9-80c8-671e8dfd260b.png)

This should work... let me know.

**Comment by Llewellynvdm** [comment]
Oops I tried that and it failed... 
![image](https://user-images.githubusercontent.com/5607939/54620660-2d19c680-4a6f-11e9-80de-00724b82b55d.png)

This because the path `[[[component]]]` was not updated. I will add that fix in there... to get that to work so it will natuarly work in any project/component and not be locked into just one.

**Comment by TLWebdesign** [comment]
> To also use it to add php stuff is not ideal, I would much rather custom code this into the view and add the files via the file and folder area.

Ahh ok. Yeah i was just thinking. Isn't it impossible to call those classes from the view adding them as libraries. Seems like it would need custom code into view anyway. Or am is missing something here?

**Comment by Llewellynvdm** [comment]
Well the libraries, not only add the files it is also used to add the files to the page. So yes the libraries could do all that, but can also be done with the normal file, folders area of a component. Which work very much the same as the library area of adding file and folders.

**Comment by TLWebdesign** [comment]
> This because the path `[[[component]]]` was not updated. I will add that fix in there... to get that to work so it will natuarly work in any project/component and not be locked into just one.

I tried the latest staging branch and it worked with some customisations. At first I also got the same error like you but that is to be expected not? The place where JCB needs to find the files in order to add them to the component is not in the com_[[[component]]]/helpers/OdrApi but inside JPATH_ADMINISTRATOR/com_componentbuilder/custom/OdrApi and then add them to site/helpers, correct? 

I have these settings in advanced tab and they work. 

![Schermafbeelding 2019-03-20 om 10 40 04](https://user-images.githubusercontent.com/4402824/54674431-b4f9e200-4afc-11e9-95a5-e5fed4242e3d.png)

On the library edit page in the Linked Files, Folders & URLs TAB it says there are no linked files or folders like this:
![Schermafbeelding 2019-03-20 om 10 33 48](https://user-images.githubusercontent.com/4402824/54673988-c5f62380-4afb-11e9-94d3-075fdff53d66.png)

The files are added on installation tho. I don't know if it is supposed to say that there's no files or folders linked. 

Also no need for the com_[[[component]]] CONSTANT JPATH_COMPONENT_SITE does the trick also like this: 		require_once( JPATH_COMPONENT_SITE.'/helpers/OdrApi/Odr.php' );

Thanks for the help!

**Comment by TLWebdesign** [comment]
only custom script is working atm tho. Don't know if you tried any fix for that yet. Using always add doesn't add the code to the view.html.php file at all anymore using advanced tab.

**Comment by TLWebdesign** [comment]
On above comment. The latest staging does show the linked fullpath from advanced tab. I mixed my installations up when installing latest staging branch. So it works now only if i use custom script and add the require_once myself like you said. Tried other combinations but those just don't add the files to the view.html.php or they add it without the right constant in front of it. For now i got it working tho thanks!
